flight/mcu/lib/teensyShared/kalmanFilter.cpp

```diff
@@ -0,0 +1,167 @@
+
+#include "kalmanFilter.h"
+#define EIGEN_MATRIX_PLUGIN "MatrixAddons.h"
+
+KalmanFilter::KalmanFilter(struct pointers* pointer_struct) {
+    gz_L = &pointer_struct->sensorDataPointer->lowG_data.gz;
+    gz_H = &pointer_struct->sensorDataPointer->highG_data.hg_az;
+    b_alt = &pointer_struct->sensorDataPointer->barometer_data.altitude;
+    mutex_lowG_ = &pointer_struct->dataloggerTHDVarsPointer.dataMutex_lowG;
+    mutex_highG_ = &pointer_struct->dataloggerTHDVarsPointer.dataMutex_highG;
+    dataMutex_barometer_ =
+        &pointer_struct->dataloggerTHDVarsPointer.dataMutex_barometer;
+    dataMutex_state_ =
+        &pointer_struct->dataloggerTHDVarsPointer.dataMutex_state;
+    stateData_ = &pointer_struct->sensorDataPointer->state_data;
+    data_logger_ = &pointer_struct->dataloggerTHDVarsPointer;
+    current_state_ =
```

Comment:
where do we use `current_state_` in this file?

Reply:
kfTickFunction(). the kf update is only done if we've passed IDLE